e2e/cat_test.go

```diff
@@ -305,4 +322,215 @@ func TestCatByVersionID(t *testing.T) {
+		0: equals(`ERROR "cat --version-id=%v s3://%v/": wildcard/prefix operations are disabled with --version-id flag`, version, bucket),
+	}, strictLineCheck(false))
+
+	cmd = s5cmd("cat", "--version-id", version, "s3://"+bucket+"/folder/")
+	result = icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Expected{ExitCode: 1})
+	assertLines(t, result.Stderr(), map[int]compareFunc{
+		0: equals(`ERROR "cat --version-id=%v s3://%v/folder/": wildcard/prefix operations are disabled with --version-id flag`, version, bucket),
+	}, strictLineCheck(false))
+
+	cmd = s5cmd("cat", "--version-id", version, "s3://"+bucket+"/*")
+	result = icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Expected{ExitCode: 1})
+	assertLines(t, result.Stderr(), map[int]compareFunc{
+		0: equals(`ERROR "cat --version-id=%v s3://%v/*": wildcard/prefix operations are disabled with --version-id flag`, version, bucket),
+	}, strictLineCheck(false))
+}
+
+func TestCatPrefix(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		files    []string
+		contents []string
+		prefix   string
+		expected string
+	}{
+		{
+			files:    []string{"file1.txt", "file2.txt"},
+			contents: []string{"content0", "content1"},
+			expected: "content0content1"},
+		{
+			files:    []string{"file1.txt", "file2.txt", "dir/file3.txt", "dir/file4.txt"},
+			contents: []string{"content0", "content1", "content2", "content3"},
+			expected: "content0content1",
+		},
+		{
+			files:    []string{"file1.txt", "file2.txt", "dir/file3.txt", "dir/file4.txt"},
+			contents: []string{"content0", "content1", "content2", "content3"},
+			prefix:   "dir/",
+			expected: "content2content3",
+		},
+		{
+			files:    []string{"file1.txt", "file2.txt", "dir/file3.txt", "dir/file4.txt", "dir/nesteddir/file5.txt"},
+			contents: []string{"content0", "content1", "content2", "content3", "content4"},
+			prefix:   "dir/",
+			expected: "content2content3",
+		},
+		{
+			files:    []string{"file1.txt", "file2.txt", "dir/file3.txt", "dir/file4.txt", "dir/nesteddir/file5.txt"},
+			contents: []string{"content0", "content1", "content2", "content3", "content4"},
+			prefix:   "dir/nesteddir/",
+			expected: "content4",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.expected, func(t *testing.T) {
+			t.Parallel()
+
+			s3client, s5cmd := setup(t)
+
+			bucket := s3BucketFromTestName(t)
+			createBucket(t, s3client, bucket)
+
+			for idx, file := range tc.files {
+				content := tc.contents[idx]
+				putFile(t, s3client, bucket, file, content)
+			}
+
+			cmd := s5cmd("cat", fmt.Sprintf("s3://%v/%v", bucket, tc.prefix))
+			result := icmd.RunCmd(cmd)
+
+			result.Assert(t, icmd.Success)
+			assertLines(t, result.Stdout(), map[int]compareFunc{
+				0: equals(tc.expected),
+			}, alignment(true))
+		})
+	}
+}
+
+func TestCatWildcard(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	files := []struct {
+		key     string
+		content string
```

Comment:
could not we use `map[string]string` here?

Reply:
Used `map[string]string` without any struct type.